components/feature/accounts-push/src/main/java/mozilla/components/feature/accounts/push/FxaPushSupportFeature.kt

```diff
@@ -104,6 +136,40 @@ internal class ConstellationObserver(
     }
 }
 
+/**
+ * An [AutoPushFeature] observer to handle [FxaAccountManager] subscriptions and push events.
+ */
+internal class PushObserver(
+    private val accountManager: FxaAccountManager
+) : Bus.Observer<PushType, String>, PushSubscriptionObserver {
+
+    private val logger = Logger("PushObserver")
+
+    override fun onSubscriptionAvailable(subscription: AutoPushSubscription) {
+        logger.debug("Received new push subscription from $subscription.type")
+
+        if (subscription.type == PushType.Services) {
+            accountManager.withConstellation {
+                it.setDevicePushSubscriptionAsync(
```

Comment:
Is it worth checking here if we already have the same subscription on the device record? That'd save us a network call, at least. It would also probably indicate some kind of a bug, which would be useful to track (e.g. via CrashReporter).

Reply:
The rust component gives us this security. If there already exists a subscription on our local db, we don't make the network call.